File: src/com/razie/pub/actionables/Actionable.java
package com.razie.pub.actionables;

import razie.base.ActionItem;
import razie.base.AttrAccess;

/**
 * needed a base class for the actionable concept
 * 
 * TODO 1-1 detailed docs
 * 
 * @author razvanc
 */
public class Actionable {

	public String name;
	public ActionItem item;
	public AttrAccess inArgs;
	public IExecutable executable;

	public Actionable(String name, ActionItem ai, AttrAccess in,
			IExecutable ex) {
		this.name = name;
		this.item = ai;
		this.inArgs = in;
		this.executable = ex;
	}

	public AttrAccess execute() {
		return this.executable.execute(inArgs);
	}
}

File: src/razie/actionables/Actionable.scala
package razie.actionables;

import razie.base.ActionContext;
import razie.base.ActionItem;
import razie.base.AttrAccess;

/**
 * needed a base class for the actionable concept
 * 
 * TODO 1-1 detailed docs
 * 
 * @author razvanc
 */
class Actionable (
      val name:String, 
      val item:ActionItem, 
      val inArgs:AttrAccess, 
      val executable:IExecutable) {
   
	def execute () : Any = {
		this.executable.apply(inArgs, null);
	}
	
	def jexecutable() = executable
	def jinArgs() = inArgs
}
